File: frontend/src/models/Account.ts
export interface Account {
  accountID: string;
  accountName?: string;
  gw2AccountName?: string;
  apiKey?: string;
  password?: string;
  sessionID?: string;
  session?: Session;
}

export interface APIAccount {
  id: string;
  account_name?: string;
  name?: string;
  gw2_name?: string;
  api_key?: string;
  password?: string;
  session_id?: string;
}

export function APIAccountToAccount(apiAccount: APIAccount): Account {
  return {
    accountID: apiAccount.id,
    accountName: apiAccount.account_name,
    gw2AccountName: apiAccount.gw2_name,
    apiKey: apiAccount.api_key,
    password: apiAccount.password,
    sessionID: apiAccount.session_id,
  };
}

interface Session {
  sessionID: string;
  expires: Date;
}

export default Account;

File: backend/internal/api/models/account.go
package models

import "time"

type Account struct {
	AccountID      string     `json:"id"`
	LastCrawl      *time.Time `json:"last_crawl"`
	AccountName    *string    `json:"name"`
	GW2AccountName *string    `json:"gw2_name"`
	APIKey         *string    `json:"api_key"`
	Password       *string    `json:"password"`
	SessionID      *string    `json:"session_id"`
	Session        *Session   `json:"session"`
}
